src/geode/mesh/builder/surface_mesh_builder.cpp

```diff
@@ -366,6 +395,7 @@ namespace geode
                         polygons_around[1], polygon_vertex_id );
                 }
             }
+            reset_polygons_around_vertex( polygon_vertex_id );

```

Comment:
Do not confuse polygon_around_vertex() with polygon_vertex(), here we want to reset the value because we are changing the polygon vertex of the polygon